src/io/flutter/inspector/FPSDisplay.java

```diff
@@ -0,0 +1,370 @@
+    labelsPanel.add(Box.createHorizontalGlue());
+    labelsPanel.add(elapsedLabel);
+
+    final FlutterFramesMonitor.Listener listener = event -> {
+      fpsPanel.update();
+
+      final int ms = Math.round(event.elapsedMicros / 1000.0f);
+      elapsedLabel.setText(ms + "ms");
+      SwingUtilities.invokeLater(elapsedLabel::repaint);
+
+      fpsLabel.setText(df.format(flutterFramesMonitor.getFPS()) + " FPS");
+      SwingUtilities.invokeLater(fpsLabel::repaint);
+    };
+
+    flutterFramesMonitor.addListener(listener);
+    Disposer.register(parentDisposable, () -> flutterFramesMonitor.removeListener(listener));
+
+    return panel;
+  }
+
+  public static class ToolbarComponentAction extends AnAction implements CustomComponentAction {
+    // The width of the drawable fps area.
+    private static final int HEAP_GRAPH_WIDTH = 60;
+
+    final @NotNull Disposable parent;
+    final @NotNull FlutterApp app;
+
+    public ToolbarComponentAction(@NotNull Disposable parent, @NotNull FlutterApp app) {
+      this.parent = parent;
+      this.app = app;
+    }
+
+    @Override
+    public void update(AnActionEvent e) {
+      // No-op.
+    }
+
+    @Override
+    public void actionPerformed(AnActionEvent e) {
+      // No-op.
+    }
+
+    @Override
+    public JComponent createCustomComponent(Presentation presentation) {
+      // Summary label.
+      final JBLabel fpsLabel = new JBLabel();
+      fpsLabel.setFont(UIUtil.getLabelFont(UIUtil.FontSize.SMALL));
+      fpsLabel.setForeground(UIUtil.getLabelDisabledForeground());
+      //fpsLabel.setPreferredSize(new Dimension(75, -1));
+
+      assert app.getPerfService() != null;
+      final FlutterFramesMonitor flutterFramesMonitor = app.getPerfService().getFlutterFramesMonitor();
+
+      // Graph component.
+      final FPSPanel fpsPanel = new FPSPanel(flutterFramesMonitor);
+      fpsPanel.setPreferredSize(new Dimension(HEAP_GRAPH_WIDTH, -1));
+
+      final FlutterFramesMonitor.Listener listener = event -> {
+        fpsPanel.update();
+
+        fpsLabel.setText(df.format(flutterFramesMonitor.getFPS()) + " FPS");
+        SwingUtilities.invokeLater(fpsLabel::repaint);
+      };
+
+      flutterFramesMonitor.addListener(listener);
+      Disposer.register(parent, () -> flutterFramesMonitor.removeListener(listener));
+
+      final JPanel container = new JPanel(new BorderLayout(5, 0));
+      container.add(fpsLabel, BorderLayout.WEST);
+      container.add(fpsPanel, BorderLayout.CENTER);
+      return container;
+    }
+  }
+}
+
+class FPSPanel extends JPanel {
+  private final FlutterFramesMonitor framesMonitor;
+
+  private final Map<FlutterFramesMonitor.FlutterFrameEvent, JComponent> frameWidgets = new HashMap<>();
+
+  private Rectangle lastSavedBounds;
+
+  FPSPanel(FlutterFramesMonitor framesMonitor) {
+    this.framesMonitor = framesMonitor;
+
+    setLayout(null);
+    final Color color = UIUtil.getLabelDisabledForeground();
+    //noinspection UseJBColor
```

Comment:
How does this look w/ a light theme?

Reply:
It works - the label color does the right thing here; I'll update w/ a screenshot.